test/Mobile/test_mobile.py

```diff
@@ -0,0 +1,220 @@
+        MobileLibrary.scroll_up_to_text(mock_desk, "some_locator")
+        mock_desk.swipe_by_percent.assert_called()
+
+    def test_scroll_up_to_text_failure(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        mock_desk._is_text_present.side_effect = [False, False, False]
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        self.assertRaises(AssertionError, MobileLibrary.scroll_up_to_text, mock_desk, "some_locator", 2)
+
+    def test_scroll_down_to_text(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        mock_desk._is_text_present.side_effect = [False, True]
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        MobileLibrary.scroll_down_to_text(mock_desk, "some_locator")
+        mock_desk.swipe_by_percent.assert_called()
+
+    def test_scroll_down_to_text_failure(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        mock_desk._is_text_present.side_effect = [False, False, False]
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        self.assertRaises(AssertionError, MobileLibrary.scroll_down_to_text, mock_desk, "some_locator", 2)
+
+    def test_wait_for_and_tap(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        MobileLibrary.wait_for_and_tap(mock_desk, "some_locator")
+        mock_desk.tap.assert_called_with("some_locator", None, None, 1)
+
+    def test_wait_for_and_tap_multiple(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        MobileLibrary.wait_for_and_tap(mock_desk, "some_locator", count=4)
+        mock_desk.tap.assert_called_with("some_locator", None, None, 4)
+
+    def test_wait_for_and_tap_override_defaults(self):
+        mock_desk = MagicMock()
+        webdriver.Remote = WebdriverRemoteMock
+        MobileLibrary.open_application(mock_desk, 'remote_url')
+        MobileLibrary.wait_for_and_tap(mock_desk, "some_locator", x_offset=200, y_offset=100, count=4,
+                                       error='some_error', timeout='10s')
+        mock_desk.tap.assert_called_with("some_locator", 200, 100, 4)
+
+    def test_capture_page_screenshot(self):
+        mock_desk = MagicMock()
+        mock_desk._get_screenshot_paths = MagicMock(return_value=['path', 'link'])
+        webdriver.Remote = WebdriverRemoteMock
+        MobileLibrary.capture_page_screenshot(mock_desk)
+        mock_desk._get_screenshot_paths.assert_called()
+
+    def test_capture_page_screenshot_else_case(self):
+        mock_desk = MagicMock()
+        mock_desk._get_screenshot_paths = MagicMock(return_value=['path', 'link'])
+        webdriver.Remote = WebdriverRemoteMock
+        del mock_desk._current_application().get_screenshot_as_file
```

Comment:
This one was pretty fun. So I needed to pull in 'Capture Page Screenshot' since the appium version didn't return the file. This lead to an 'else' statement that i had issues getting covered since it was using the builtin 'getattr' methods that if mocked caused a ton of issues since its a base building block of python.

So this came in handy as it allowed me to remove that attribute it was looking for in my mock specifically to cause that else statement to happen. https://docs.python.org/3/library/unittest.mock.html#deleting-attributes